dashboards/kubelet.libsonnet

```diff
@@ -17,7 +17,7 @@ local singlestat = grafana.singlestat;
           span=2,
           valueName='min',
         )
-        .addTarget(prometheus.target('sum(up{%(clusterLabel)s="$cluster", %(kubeletSelector)s})' % $._config));
+        .addTarget(prometheus.target('sum(up{%(clusterLabel)s="$cluster", %(kubeletSelector)s}, metrics_path="/metrics")' % $._config));
```

Comment:
@paulfantom @brancz this change breaks kube-prometheus.
https://github.com/prometheus-operator/kube-prometheus/blob/07136d1d6e0995908f3b418e4d8337cf769e1006/jsonnet/kube-prometheus/components/k8s-control-plane.libsonnet#L13 already has kubeletSelector: 'job="kubelet", metrics_path="/metrics"' defined. Also, above introduces a syntactical error in prom query expression.

Reply:
> this change breaks kube-prometheus.

Then it is kube-prometheus issue and should be fixed there. Do you want to create a PR?

>  Also, above introduces a syntactical error in prom query expression

Can you point where you see the error?

Reply:
@paulfantom  above change generates
` "expr": "sum(up{cluster=\"$cluster\", job=\"kubelet\", metrics_path=\"/metrics\"}, metrics_path=\"/metrics\")",`

1.) Note the metrics_path is outside the `{}` expression.  That is the syntax error.

2.) since kube-prometheus default config defines `kubeletSelector: 'job="kubelet", metrics_path="/metrics"'`, do you still need to metrics_path directly defined here ? Why can't  it be captured as part of config variable `kubeSelector` as it is now ?

Reply:
I added some changes in #601 to fix syntax error and switch to different metric being used.

---

> since kube-prometheus default config defines kubeletSelector: 'job="kubelet", metrics_path="/metrics"', do you still need metrics_path directly defined here ?

kubernetes-mixin is not meant to be used solely with kube-prometheus, if it were, it would be in kube-prometheus and not as a separate project. This said, we had a bug in the query here and had to solve it here and not in a project which depends on kubernetes-mixin.

Also bear in mind, that promql allows having duplicate label selectors in query, so the following query is valid and should return data:
```
sum(up{job="kubelet", metrics_path="/metrics", metrics_path="/metrics"})
```